packages/amplify-provider-awscloudformation/src/graphql-transformer/utils.ts

```diff
@@ -4,37 +4,21 @@ import { TransformerProjectConfig, DeploymentResources } from '@aws-amplify/grap
 import rimraf from 'rimraf';
 import { ProviderName as providerName } from '../constants';
 import { $TSContext, AmplifyCategories, JSONUtilities, pathManager, stateManager } from 'amplify-cli-core';
-import { CloudFormation, Template, Fn } from 'cloudform';
-import { Diff, diff as getDiffs } from 'deep-diff';
+import { CloudFormation, Fn } from 'cloudform';
 import { ResourceConstants } from 'graphql-transformer-common';
 import { pullAllBy, find } from 'lodash';
 import { isAmplifyAdminApp } from '../utils/admin-helpers';
 import { printer } from 'amplify-prompts';
 import { prePushCfnTemplateModifier } from '../pre-push-cfn-processor/pre-push-cfn-modifier';
 
-const ROOT_STACK_FILE_NAME = 'cloudformation-template.json';
 const PARAMETERS_FILE_NAME = 'parameters.json';
 const CUSTOM_ROLES_FILE_NAME = 'custom-roles.json';
 const AMPLIFY_ADMIN_ROLE = '_Full-access/CognitoIdentityCredentials';
 const AMPLIFY_MANAGE_ROLE = '_Manage-only/CognitoIdentityCredentials';
 
-export interface CustomRolesConfig {
+interface CustomRolesConfig {
   adminRoleNames?: Array<string>;
 }
-export interface DiffableProject {
-  stacks: {
-    [stackName: string]: Template;
-  };
-  root: Template;
-}
-
-export type DiffChanges<T> = Array<Diff<DiffableProject, DiffableProject>>;
-
-export interface GQLDiff {
-  diff: DiffChanges<DiffableProject>;
-  next: DiffableProject;
-  current: DiffableProject;
-}
 
 export const getIdentityPoolId = async (ctx: $TSContext): Promise<string | undefined> => {
```

Comment:
I don't have complete context on this. Should we move `getIdentityPoolId` and `writeDeploymentToDisk` to resource-manager folder?

Reply:
These utilities are both used downstream of `transformGraphQLSchema`, which I'm planning to port over to our library. We're `transformGraphQLSchema` is pretty tightly coupled to the provider in general, but I think that the refactor effort for our GQL compilation should address this a bit better. We'll still be somewhat tightly coupled, but I'd bias towards having that coupling live in our domain for the time being than live in the provider.

Reply:
Chatted offline, but we're planning to leave as-if for this commit, since this is actively being refactored. The utilities are basically remaining in the package which they have the most in common with, and we should be addressing this in a separate commit.